packages/lab/src/Flow/hooks/useBaseNode.tsx

```diff
@@ -0,0 +1,192 @@
+import { isValidElement, useCallback, useMemo, useState } from "react";
+import { NodeToolbarProps } from "reactflow";
+import { uid } from "uid";
+import { useLabels } from "@hitachivantara/uikit-react-core";
+import { Delete, Duplicate } from "@hitachivantara/uikit-react-icons";
+import { getColor, HvColorAny, theme } from "@hitachivantara/uikit-styles";
+
+import { useNodeMetaRegistry } from "../FlowContext/NodeMetaContext";
+import { identifyHandles } from "../Node/utils";
+import {
+  HvFlowNodeAction,
+  HvFlowNodeInput,
+  HvFlowNodeInputGroup,
+  HvFlowNodeOutput,
+  HvFlowNodeOutputGroup,
+} from "../types";
+import { useFlowContext } from "./useFlowContext";
+import { useFlowInstance } from "./useFlowInstance";
+import {
+  useFlowNode,
+  useFlowNodeInputEdges,
+  useFlowNodeOutputEdges,
+} from "./useFlowNode";
+
+export const DEFAULT_LABELS = {
+  deleteActionLabel: "Delete",
+  duplicateActionLabel: "Duplicate",
+};
+
+export interface HvFlowUseBaseNodeProps {
+  id?: string;
+  /** Node group ID */
+  groupId?: string;
+
+  title?: string;
+
+  subtitle?: string;
+
+  icon?: React.ReactNode;
+
+  color?: HvColorAny;
+  /** Node inputs */
+  inputs?: (HvFlowNodeInput | HvFlowNodeInputGroup)[];
+  /** Node outputs */
+  outputs?: (HvFlowNodeOutput | HvFlowNodeOutputGroup)[];
+  /** Node actions */
+  nodeActions?: HvFlowNodeAction[];
+  /** Labels used on the default actions. */
+  labels?: Partial<typeof DEFAULT_LABELS>;
+  /** Props for the NodeToolbar Component */
+  nodeToolbarProps?: NodeToolbarProps;
+}
+
+export function useHvBaseNode(props: HvFlowUseBaseNodeProps) {
```

Comment:
shouldn't we be leveraging this in `HvBaseNode`?

also, this hook isn't being exported